lib/transformer.dart

```diff
@@ -46,52 +46,82 @@ abstract class DoughTransformer {
   /// The controller for the associated [Dough] widget.
   DoughController get controller => _controller;
 
-  /// Create the matrix which will be used to transform the [Dough.child] widget.
-  vmath.Matrix4 createDoughMatrix();
+  /// Creates the [Matrix4] which will be used to transform the [Dough.child] widget.
+  Matrix4 createDoughMatrix();
 
-  /// A utility method which returns a matrix that scales widgets.
+  /// A utility method which creates a [Matrix4] that scales widgets by a factor of
+  /// [DoughRecipe.expansion].
   @protected
-  vmath.Matrix4 expansionMatrix() {
-    // TODO use a homography here to scale non-uniformly?
+  Matrix4 createExpansionMatrix() {
+    // TODO:
+    // Try to recreate photoshop's liquify effect to push pixels closest to the press
+    // point (target) away (1/x). This could give illusion that the screen is squishy
+    // dough.
     final scaleMag = ui.lerpDouble(1, recipe.expansion, t);
-    return Matrix4.identity()..scale(scaleMag, scaleMag, scaleMag);
+    return Matrix4.identity()..scale(scaleMag);
   }
 
-  /// A utility method which returns a matrix that rotates widgets in the direction
-  /// of the [deltaAngle] property.
+  /// A utility method which creates a [Matrix4] that perspectively rotates wigets around
+  /// their yaw and pitch axes based on [delta] and [DoughRecipe.viscosity].
   @protected
-  vmath.Matrix4 rotateTowardDeltaMatrix() {
-    return Matrix4.rotationZ(deltaAngle);
-  }
+  Matrix4 createPerspectiveWarpMatrix() {
+    if (!recipe.usePerspectiveWarp) {
+      return Matrix4.identity();
+    }
 
-  /// A utility method which returns a matrix that rotates widgets int the opposite
-  /// direction of the [deltaAngle] property.
-  @protected
-  vmath.Matrix4 rotateAwayFromDeltaMatrix() {
-    return Matrix4.rotationZ(-deltaAngle);
+    final perspDelta = -delta * t / recipe.viscosity;
```

Comment:
Only using viscosity here because adhesion relates to stickiness, viscosity is liquidness